tests/testthat/test-utils.R

```diff
@@ -284,13 +293,234 @@ test_that("longwide works as expected with default values", {
+    sub_group <- sub_syn[sub_syn$gcr_result == "Include",]
+    sum(table(sub_group$agedays[sub_group$subjid == i]) >= 2)
+  })
+
+  # it should be that the number of occurrences for each subject in "all_obs" is
+  # the same as in "wide_syn", so check that but make sure sorted by subjid
+  ws_subj_counts <- table(wide_syn$subjid)
+  ws_subj_counts <- setNames(as.vector(ws_subj_counts), names(ws_subj_counts))
+
+  expect_equal(all_obs[sort(names(all_obs))],
+               ws_subj_counts[sort(names(ws_subj_counts))])
+
+  # get all the observation ID's in the wide data
+  obs_ids <- c(wide_syn$wt_id, wide_syn$ht_id)
+
+  # here specified inclusion type is just "include"
+  # check that it includes specified inclusion types
+  expect(
+    all(sub_syn$gcr_result[sub_syn$id %in% obs_ids] == "Include"),
+    "longwide() includes inclusion values that were not specified"
+  )
+
+  # check that all sexes have been correctly recoded
+  orig_sex <- sub_syn[!duplicated(sub_syn$subjid), c("subjid", "sex")]
+  orig_sex <- orig_sex[order(orig_sex$subjid),]
+
+  aft_sex <- wide_syn[!duplicated(wide_syn$subjid), c("subjid", "sex")]
+  aft_sex <- aft_sex[order(aft_sex$subjid),]
+
+  expect_equal(orig_sex$sex+1, aft_sex$sex)
+
+  # spot check that data is correct; add in checks for additional columns
+  set.seed(7)
+  # check height ids
+  ht_sub <-
+    sub_syn[sub_syn$param == "HEIGHTCM" & sub_syn$id %in% obs_ids, ]
+  for (x in ht_sub$id[sample(1:nrow(ht_sub), 20)]) {
+    w_idx <- wide_syn$ht_id == x
+    ht_idx <- ht_sub$id == x
+
+    # check ages
+    expect_equal(wide_syn$agey[w_idx], round(ht_sub$agedays[ht_idx] / 365.25), 4)
+    expect_equal(wide_syn$agem[w_idx],
+                 round(round(ht_sub$agedays[ht_idx] / 365.25), 4) * 12, 4)
+    expect_equal(wide_syn$agedays[w_idx], ht_sub$agedays[ht_idx])
+
     # check height
+    expect_equal(wide_syn$ht[w_idx], ht_sub$measurement[ht_idx])
+
+    # check extra columns
+    expect_equal(wide_syn$ht_r1[w_idx], ht_sub$r1[ht_idx])
+    expect_equal(wide_syn$r2[w_idx], ht_sub$r2[ht_idx])
+  }
+
+  # check weight ids
+  wt_sub <-
+    sub_syn[sub_syn$param == "WEIGHTKG" & sub_syn$id %in% obs_ids, ]
+  for (x in wt_sub$id[sample(1:nrow(wt_sub), 20)]) {
+    w_idx <- wide_syn$wt_id == x
+    wt_idx <- wt_sub$id == x
+
+    # check ages
+    expect_equal(wide_syn$agey[w_idx], round(wt_sub$agedays[wt_idx] / 365.25), 4)
+    expect_equal(wide_syn$agem[w_idx],
+                 round(round(wt_sub$agedays[wt_idx] / 365.25), 4) * 12, 4)
+    expect_equal(wide_syn$agedays[w_idx], wt_sub$agedays[wt_idx])
+
+    # check weight
     expect_equal(wide_syn$wt[w_idx], wt_sub$measurement[wt_idx])
+
+    # check extra columns
+    expect_equal(wide_syn$wt_r1[w_idx], wt_sub$r1[wt_idx])
+    expect_equal(wide_syn$r2[w_idx], wt_sub$r2[wt_idx])
+  }
+})
+
+test_that("longwide works as expected when not dropping unmatched values", {
+  # use synthetic data, running cleaning on a subset
```

Comment:
A lot of this code is copied from above -- could the two tests be merged?

Reply:
I went ahead and dropped the spot checks from this test. They are somewhat redundant.